includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -665,6 +690,54 @@ function ( $value ) {
 		return compact( 'stylesheet', 'validation_errors' );
 	}
 
+	/**
+	 * Add placeholders for calc() functions which the PHP-CSS-Parser doesn't handle them properly yet.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/sabberworm/PHP-CSS-Parser/issues/79
+	 *
+	 * @param string $css CSS.
+	 * @return string CSS with calc() functions replaced with placeholders.
+	 */
+	private function add_calc_placeholders( $css ) {
+		$offset = 0;
+		while ( preg_match( '/(?:-\w+-)?\bcalc\(/', $css, $matches, PREG_OFFSET_CAPTURE, $offset ) ) {
+			$match_string = $matches[0][0];
+			$match_offset = $matches[0][1];
+			$css_length   = strlen( $css );
+			$open_parens  = 1;
+			$start_offset = $match_offset + strlen( $match_string );
+			$final_offset = $start_offset;
+			for ( ; $final_offset < $css_length; $final_offset++ ) {
+				if ( '(' === $css[ $final_offset ] ) {
+					$open_parens++;
+				} elseif ( ')' === $css[ $final_offset ] ) {
+					$open_parens--;
+				}
```

Comment:
Probably should add a case here to catch `;` or `}` and `break` because clearly that indicates unbalanced parentheses and we should skip to looking for the next match from that point forward.

Reply:
It looks like the `elseif` block below now handles that case.